### DISCUSSION TITLE
A way to easily publish frontend assets

### DISCUSSION THREAD

**Discussion by ArnaudLier** [filamentphp/filament#3209] [Ideas]
Being able to publish frontend assets [just like Livewire](https://laravel-livewire.com/docs/2.x/installation#publish-assets) would be great for people that doesn't want assets to be served through Laravel but directly by the web server.

**Comment by danharrin** [comment]
What is your objection to serving them from the web server?

I am incredibly opposed to adding this function like Livewire, it causes so many unnecessary support requests from people who don't know how to read documentation & don't republish the assets after every update

**Reply by ArnaudLier** [reply]
Performance, mainly. It makes no sense to fire up Laravel if the webserver can just serve the files in my opinion.

Through Laravel: Requests per second:    63.45 [#/sec] (mean)
Directly by the web server: Requests per second:    697.33 [#/sec] (mean)

**Reply by wsamoht** [reply]
I was just looking on how to do this as well. My use case is that I am going to be using Laravel Vapor and need the assets to be served via the CDN and not Lambda. Is there any work around atm?

**Reply by danharrin** [reply]
I don't know for v2 but in v3 all assets are published. Vapor users have definitely got Filament to work before

**Reply by wsamoht** [reply]
Awesome! We can wait until v3. Right now v2 works, the JS/CSS just gets served via Lambda.
